### PR TITLE
[Feature](profile) support ProfilePlugin

### PR DESCRIPTION
**Pull request by dutyu**
Add `ProfilePlugin` support, when ProfileManager receive a profile, send this profile to ProfileEventProcessor and handle it if we have installed some ProfilePlugin(s). So we can build a `ProfileLoader` plugin to persist profiles to doris.

# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.06 seconds
 load time: 471 seconds
 storage size: 17171800834 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230206030018_clickbench_pr_90595.html

File: fe/fe-core/src/main/java/org/apache/doris/common/util/ProfileManager.java [hunk 1]
```diff
@@ -165,6 +168,14 @@ public void pushProfile(RuntimeProfile profile) {
         }
 
         ProfileElement element = createElement(profile);
+
+        ProfileEventProcessor processor = Env.getCurrentProfileEventProcessor();
+        // we don't need to handle running profile
+        // we don't need to handle it if there isn't any installed ProfilePlugin
```
Comment (carlvinhust2012) [inline fe/fe-core/src/main/java/org/apache/doris/common/util/ProfileManager.java:174]:
```suggestion
        // we don't need to handle the running profile if there isn't any installed ProfilePlugin
```

File: fe/fe-core/src/main/java/org/apache/doris/qe/AbstractEventProcessor.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+                    LOG.debug("encounter exception when processing event.", e);
+                }
+
+            }
+        }
+
+    }
+
+    // should return a not null value
+    protected abstract @NotNull List<Plugin> getPlugins();
```
Comment (carlvinhust2012) [inline fe/fe-core/src/main/java/org/apache/doris/qe/AbstractEventProcessor.java:143]: why not put these functions to the front of the file?

**Comment by dutyu** [comment]
run buildall

**Pull request closed without merging**